File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0020-vacation-days/ExtraDaysCase3RuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "ExtraDaysCase3"})
public class ExtraDaysCase3RuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal extraDaysCase3;

    public ExtraDaysCase3RuleOutput(boolean matched) {
        super(matched);
    }

    public java.math.BigDecimal getExtraDaysCase3() {
        return this.extraDaysCase3;
    }
    public void setExtraDaysCase3(java.math.BigDecimal extraDaysCase3) {
        this.extraDaysCase3 = extraDaysCase3;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ExtraDaysCase3RuleOutput other = (ExtraDaysCase3RuleOutput) o;
        if (this.getExtraDaysCase3() != null ? !this.getExtraDaysCase3().equals(other.getExtraDaysCase3()) : other.getExtraDaysCase3() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getExtraDaysCase3() != null ? this.getExtraDaysCase3().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", extraDaysCase3='%s'", extraDaysCase3));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0020-vacation-days/ExtraDaysCase3RuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "ExtraDaysCase3"])
class ExtraDaysCase3RuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("ExtraDaysCase3")
    var extraDaysCase3: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ExtraDaysCase3RuleOutput
        if (if (this.extraDaysCase3 != null) this.extraDaysCase3 != other.extraDaysCase3 else other.extraDaysCase3 != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.extraDaysCase3 != null) this.extraDaysCase3.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", extraDaysCase3='%s'", extraDaysCase3))
        result_.append(")")
        return result_.toString()
    }
}
